internal/resources/disclaimer_annotation.go

```diff
@@ -0,0 +1,34 @@
+package resources
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DisclaimerKey   = "istios.operator.kyma-project.io/managed-by-disclaimer"
+	DisclaimerValue = "DO NOT EDIT - This resource is managed by Kyma.\nAny modifications are discarded and the resource is reverted to the original state."
+)
+
+// AnnotateWithDisclaimer adds the managed-by-disclaimer annotation to the given resource and updates it in the cluster.
+func AnnotateWithDisclaimer(ctx context.Context, resource unstructured.Unstructured, k8sClient client.Client) error {
+	annotations := resource.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[DisclaimerKey] = DisclaimerValue
+	resource.SetAnnotations(annotations)
+
+	err := k8sClient.Update(ctx, &resource)
+	return err
+}
+
+func hasManagedByDisclaimer(resource unstructured.Unstructured) bool {
```

Comment:
can you please test this function as well (in `disclaimer_annotation_test.go`). you can utilize it in `apply_test.go` as well.